src/components/Posts/EditPostModal.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  Dialog,
+  DialogTitle,
+  Button,
+  DialogContent,
+  IconButton,
+  Box,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import { useEffect, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import { EmojiClickData } from "emoji-picker-react";
+import EmojiPickerIconButton from "../Common/EmojiPickerIconButton";
+import UserAvatar from "../Common/UserAvatar";
+import AddPhotoAlternateOutlinedIcon from "@mui/icons-material/AddPhotoAlternateOutlined";
+import axios from "axios";
+import { updatePost } from "../../state/slices/postsSlice";
+
+type EditPostModalProps = {
+  onClose: () => void;
+  open: boolean;
+  postId: number;
+};
+
+const styles = {
+  paperProps: {
+    overflow: "visible",
+    borderRadius: 20,
+  },
+  dialog: {
+    ".MuiDialog-scrollPaper": { alignItems: "flex-start" },
+  },
+  dialogTitle: {
+    display: "flex",
+    alignItems: "center",
+  },
+  titleText: {
+    fontWeight: "bold",
+    paddingLeft: 3,
+  },
+  dialogContentContainer: {
+    display: "flex",
+    paddingBottom: 2,
+    paddingLeft: 2,
+    paddingRight: 2,
+    paddingTop: 1,
+  },
+  avatar: { paddingRight: 1.5 },
+  textFieldContainer: { width: "100%", paddingTop: 0.5 },
+  textField: { paddingBottom: 2 },
+  postActions: {
+    display: "flex",
+    justifyContent: "space-between",
+    marginTop: 1,
+  },
+};
+
+const EditPostModal = ({ onClose, open, postId }: EditPostModalProps) => {
+  const user = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+  const existingPost = useAppSelector((state) =>
+    state.posts.posts.find((post) => post.postId === postId)
+  );
```

Comment:
Let's put the non-null assertion up here so we don't have to propagate it everywhere. This shouldn't ever really be undefined because there should always be a post with that postId. Maybe we can handle the error better if it were to ever happen but that can be an exercise to do some other time

```suggestion
  const existingPost = useAppSelector((state) =>
    state.posts.posts.find((post) => post.postId === postId)
  )!;
```

Reply:
Makes sense. Poor coding on my part.